### ISSUE TITLE
Problem with pair_basis when rin=0

### ISSUE THREAD

**Issue by wcwitt** [ACEsuit/ACE1.jl#13]
Here's a minimal example script, which works as expected.

    using ACE1    
    Bpair = pair_basis(species=[:C,:H],
                       r0=1.1,
                       maxdeg=5,
                       rcut=5.5,
                       rin=0.0,
                       pin=0)

However, changing `rin=0.0` to `rin=0` causes it to gobble up memory (at least 100GB) until the system crashes.

**Comment by casv2** [comment]
You should specify `pcut = 1` to specify to the pair basis to only have one cutoff, which would be the outer one. This way you don't specify the `rin` at all which most likely solves this problem.

**Comment by wcwitt** [comment]
Thanks, I didn't know that. But I've now found the problem persists with `rin=1`. It's the mismatch between int and float primarily.

I believe I've traced it to the construction of `TransformedPolys`, which is defined as
```
struct TransformedPolys{T, TT, TJ} <: ACE1.ScalarBasis{T}
   J::TJ          # the actual basis
   trans::TT      # coordinate transform
   rl::T          # lower bound r
   ru::T          # upper bound r = rcut
en
```
Note the definition expects `rl` and `ru` to be of the same type, `T`.

I wondered how the code compiled/ran at all if `rl` and `ru` were different types until I found this outer constructor
```
TransformedPolys(J, trans, rl, ru) =
   TransformedPolys(J, trans, rl, ru)
```
I'm not certain, but I suspect it's behavior (with two identical signatures) is undefined, or something like that. At minimum, it hides the problem. If I comment that line, the code works with float/float, but errors with int/float, which is better than catastrophe.

I can imagine several fixes, but what is the proper "Julian" way of doing things?
